File: examples/23_perl/new.pl
use strict;
use warnings;
use feature 'say';

sub greet {
    my $name = shift;
    my $greeting = shift // "Hello";
    say "$greeting, $name!";
}

sub greet_all {
    my @names = @_;
    for my $name (@names) {
        greet($name);
    }
}

greet("World");
greet("Alice", "Hi");
greet_all("Bob", "Carol", "Dave");

File: tests/examples/22_lua/new.lua
local M = {}

local default_greeting = "Hello"

function M.greet(name, greeting)
    greeting = greeting or default_greeting
    print(greeting .. ", " .. name .. "!")
end

function M.greet_table(names)
    for _, name in ipairs(names) do
        M.greet(name)
    end
end

return M
